splinterd/src/main.rs

```diff
@@ -500,13 +530,18 @@ fn setup_metrics_recorder(config: &Config) -> Result<(), UserError> {
     Ok(())
 }
 
-fn start_daemon(matches: ArgMatches) -> Result<(), UserError> {
-    // get provided config file or search default location
-    let config_file = matches
+fn get_config_file<'a>(matches: &'a ArgMatches) -> &'a str {
```

Comment:
This change could be a separate commit prior to the larger commit.

Reply:
fixed